pinot-core/src/main/java/org/apache/pinot/core/data/manager/InstanceDataManager.java

```diff
@@ -183,4 +184,11 @@ void addOrReplaceSegment(String tableNameWithType, String segmentName)
    * Immediately stop consumption and start committing the consuming segments.
    */
   void forceCommit(String tableNameWithType, Set<String> segmentNames);
+
+  /**
+   * Enables the installation of a method to determine if a server is ready to server queries.
+   *
+   * @param isServerReadyToServeQueries supplier to retrieve state of server.
+   */
+  void setSupplierOfIsServerReadyToServeQueries(Supplier<Boolean> isServerReadyToServeQueries);
```

Comment:
Added default init to avoid the problem you are describing, may be we can refine later if necessary: see commit 3ce7b783c5dc88073106f400f00783ba4ab71281

Reply:
^^